File: src/model/Expense.java
package model;
import java.time.LocalDate;
public class Expense {
    public int amount;
    public String category;
    public String description;
    public LocalDate date;
    public Expense(int amount,String category,String description){        
     
        this.amount=amount;
        this.category=category;
        this.description=description;
        this.date=LocalDate.now();  
    }
    public Expense(int amount,String category,String description,LocalDate date){       
        
        this.amount=amount;
        this.category=category;
        this.description=description;
        this.date=date;  
    }
}

File: frontend/src/types/expense.ts
/**
 * Mirrors expense_tracker.model.Expense from the backend.
 *
 * Note: `amount` is a whole-number integer on the backend (no decimals),
 * and `date` is serialized as an ISO date string ("YYYY-MM-DD").
 */
export interface Expense {
  id: number;
  amount: number;
  category: string;
  description: string;
  date: string;
}

/** Shape sent to the backend when creating or updating an expense. */
export type ExpenseInput = {
  amount: number;
  category: string;
  description: string;
  date: string;
};

/** Response shape of GET /api/expenses/summary — category name to total amount. */
export type CategorySummary = Record<string, number>;
